repose-aggregator/components/filters/rate-limiting/src/test/java/org/openrepose/filters/ratelimiting/RateLimitingHandlerTest.java

```diff
@@ -93,7 +94,12 @@ public void setup() {
             defaultConfig.setValue(10);
             defaultConfig.setUnit(org.openrepose.core.services.ratelimit.config.TimeUnit.MINUTE);
 
-            when(mockedRequest.getHeaderNames()).thenReturn(Collections.enumeration(headerNames));
+            when(mockedRequest.getMethod()).thenReturn("GET");
+            when(mockedRequest.getHeaderNames()).thenAnswer(new Answer<Object>() {
```

Comment:
The original declaration of the mock'd when/thenReturn of the `headerNames` in the `setup()` was actually non-functional. This is why at the top of each method there was another re-declaration of this when/thenReturn clause. I just pulled working when/thenReturn from the top of each method and placed it in the `setup()` in turn forcing the `final` declaration of the `headerNames`.